src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/FrameOfT.cs

```diff
@@ -113,21 +113,25 @@ public class Frame<TContext> : Frame
                                 await FireOnCompleted();
                             }
 
-                            _application.DisposeContext(context, _applicationException);
-                        }
+                            // If _requestAbort is set, the connection has already been closed.
+                            if (Volatile.Read(ref _requestAborted) == 0)
+                            {
+                                ResumeStreams();
 
-                        // If _requestAbort is set, the connection has already been closed.
-                        if (Volatile.Read(ref _requestAborted) == 0)
-                        {
-                            ResumeStreams();
```

Comment:
We should be sure to call ` _application.DisposeContext` if `messageBody.Consume()` throws. The best way to test this is to have an app func that doesn't read the request body and then send a request with malformed chunking.

I mentioned this in a previous iteration, but want to make sure it didn't get lost.